src/main/java/ToDoList.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Arrays;
+
+public class ToDoList {
+    private Task[] taskList;
+    private String[] markedTask;
+    private int taskListTracker;
+    public String checker;
+    public String marker;
+    public String unmarked;
+    public String toDo;
+
+public ToDoList(){
+    taskList = new Task[100];
+    markedTask = new String[100];
+    taskListTracker = 0;
+    checker = "list";
+    marker = "mark";
+    unmarked = "unmark";
+    toDo = "todo";
+}
+public void addTask(String incomingTask) {
+    Task newTask = new Task(incomingTask);
+    taskList[taskListTracker] = newTask;
+    taskListTracker += 1;
+    System.out.println("added: " + newTask.toBeDone);
+    System.out.println("Now you have " + taskList.length + "tasks in the list.");
+
+}
+
+public void listTask(){
+    int count = 0;
+    for(int i = 0; i < taskList.length; i +=1) {
+        if (taskList[i] == null) {
+            count += 1;
+        }
+    }
+    Task[] answer = (Arrays.copyOf(taskList,taskList.length - count));
+
+    System.out.println("Here are the tasks in your list");
+    for(int j = 0; j < answer.length; j += 1) {
+
+        if (answer[j].taskType[0] == "T") {
+            System.out.println(j + 1 + "." + Arrays.toString(answer[j].taskType) +
+                    Arrays.toString(answer[j].markAsDone) + " " + answer[j].toBeDone);
+        }
+        else if(answer[j].taskType[0] == "D"){
+            System.out.println(j + 1 + "." + Arrays.toString(answer[j].taskType) +
+                    Arrays.toString(answer[j].markAsDone)
+                    + " " + answer[j].toBeDone
+                    + " (by: " + answer[j].dueDate + ")");
+        }
+        else if (answer[j].taskType[0] == "E"){
+            System.out.println(j + 1 + "." + Arrays.toString(answer[j].taskType) +
+                    Arrays.toString(answer[j].markAsDone) + " " + answer[j].toBeDone
+                        + " (from: " + answer[j].startTime
+                        + " to: "+ answer[j].endTime + ")");
```

Comment:
Should the number of spacing be reduced?

```suggestion
                    Arrays.toString(answer[j].markAsDone) + " " + answer[j].toBeDone
                    + " (from: " + answer[j].startTime
                    + " to: "+ answer[j].endTime + ")");
```